Source/core/Thread.cpp

```diff
@@ -141,9 +151,25 @@ POP_WARNING()
 #endif
 
 #ifdef __POSIX__
-        void* Thread::StartThread(Thread* cClassPointer)
+    void* Thread::StartThread(Thread* cClassPointer)
```

Comment:
Would it not be an option to move the way Linux sets it as well here, just move the Linux here and you do not need a differentiation anymore between Linux and apple.. So also set the Process name for Linux from its own context..
Try to avoid conditional code blocks if possible without sacrificing functionality off-course.

Reply:
done